prometheus/histogram_test.go

```diff
@@ -1455,3 +1455,546 @@ func compareNativeExemplarValues(t *testing.T, exps []*dto.Exemplar, values []fl
 		}
 	}
 }
+
+func SyncMaptomap(syncmap *sync.Map) (m map[int]int64) {
+	m = map[int]int64{}
+	syncmap.Range(func(key, value any) bool {
+		m[key.(int)] = *value.(*int64)
+		return true
+	})
+	return m
+}
+
+func TestConstNativeHistogram(t *testing.T) {
```

Comment:
I'm struggling a bit to understand what is being tested here 🤔 

A new constant metric is a fairly simple operation and constant metrics do not have any methods that change their internal fields such as `Observe()`, `Add()` or `Inc()`.  It's so simple that we don't even have tests for `NewConstHistogram` 🤔 

I'm not saying tests aren't useful here, I think it's worth verifying valid labels. I'm not a native histogram expert but maybe there are other things we need to make sure are correct?

Reply:
THis test is an exact copy of existing test cases for native histograms in the same file.  https://github.com/prometheus/client_golang/blob/1eea8641d00e125557cf50c8123840608c1cba89/prometheus/histogram_test.go#L471 .  The tests ensure that native histograms serialised from an `observe()`ed nativehistogram is the same as the one created from constNativeHistograms - Or that serialisation of constNativehistogram is as it exists today .  Agree that this isn't really testing all the scenarios, I'll add more tests.